gnomad_qc/v4/create_release/create_de_novo_release.py

```diff
@@ -42,304 +31,210 @@
+    ht = ht.filter(~hl.is_missing(ht.de_novo_call_info.confidence))
+    logger.info(f"Getting {ht.count()} de novos with a confidence level...")
 
-
-def custom_an_select(ht: hl.Table, **_) -> Dict[str, hl.expr.Expression]:
-    """
-    Select AN for release.
-
-    :param ht: Hail Table with AN.
-    :return: Select expression dict.
-    """
-    selects = {"exomes_an": ht.AN, **custom_region_flags_select(ht, data_type="")}
-    selects["region_flags"] = selects["region_flags"].annotate(
-        **{r: ht[r] for r in ht.row_value if r.endswith("_region")}
+    ht = filter_low_conf_regions(
+        ht, filter_lcr=True, filter_decoy=False, filter_segdup=True
+    )
+    logger.info(
+        f"Getting {ht.count()} de novos after removing low confidence regions..."
     )
 
-    return selects
-
-
-def custom_an_select_globals(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
-    """
-    Select AN Table globals for release.
+    diploid_expr, hemi_x_expr, hemi_y_expr = get_copy_state_by_sex(
+        ht.locus,
+        ht.is_xx,
+    )
 
-    :param ht: Hail Table with AN.
-    :return: Select expression dict.
-    """
-    return {
-        "exomes_an_globals": hl.struct(
-            meta=ht.strata_meta,
-            sample_count=ht.strata_sample_count,
+    adj_trio = (
+        hl.case()
+        .when(
+            diploid_expr,
+            ht.proband_entry.adj & ht.father_entry.adj & ht.mother_entry.adj,
         )
-    }
-
+        .when(hemi_x_expr, ht.proband_entry.adj & ht.mother_entry.adj)
+        .when(hemi_y_expr, ht.proband_entry.adj & ht.father_entry.adj)
+        .or_missing()
+    )
 
-def get_de_novo_config() -> Dict[str, dict]:
-    """
-    Get de novo release config.
+    ht = ht.annotate(adj_trio=adj_trio)
+
+    ht = (
+        ht.group_by("locus", "alleles")
+        .aggregate(
+            de_novo_AC=hl.struct(
+                AC_all=hl.agg.count_where(ht.adj_trio),
+                AC_all_raw=hl.agg.count(),
+                AC_high_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "HIGH") & ht.adj_trio
+                ),
+                AC_medium_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "MEDIUM") & ht.adj_trio
+                ),
+                AC_low_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "LOW") & ht.adj_trio
+                ),
+            ),
+            p_de_novo_stats=hl.agg.stats(ht.de_novo_call_info.p_de_novo),
+        )
+        .key_by("locus", "alleles")
+    )
+    ht = ht.annotate(p_de_novo_stats=ht.p_de_novo_stats.drop("n", "sum")).checkpoint(
+        new_temp_file("denovo_ac", "ht")
+    )
+    logger.info(f"Getting {ht.count()} unique de novo variants after computing AC...")
+
+    ht = ht.annotate(vep=vep_ht[ht.key].vep)
+    ht = process_consequences(ht, has_polyphen=False)
+
+    ht = ht.annotate(
+        alt_is_star=ht.alleles[1] == "*",
```

Comment:
rather than `alt_is_star`, the annotation added earlier (`mixed_site`) might be more valuable. users can pretty easily check for themselves whether the alt is a star allele

Reply:
yep, they're different annotations -- I suggested keeping `mixed_site` in case you were trying to retain information about the locus for our users.

for `alt_is_star` specifically, it feels cleaner to filter directly using the logic here:
```
ht = ht.filter(~ht.alleles[1] == "*").checkpoint(new_temp_file("denovo_no_star", "ht"))
```
as opposed to keeping this annotation. the reason I suggest this is because the TSV will still have the `alt_is_star` annotation despite it always being False, and users that are reading in the HT can easily filter on this logic themselves as well